include/pybind11/pybind11.h

```diff
@@ -435,6 +444,16 @@ class cpp_function : public function {
                 if (options::show_function_signatures()) signatures += "\n";
             }
         }
+        if (rec->is_operator) {
+            signatures += "\n";
+            signatures += std::to_string(++index) + ". ";
+            signatures += rec->name;
+            signatures += "(";
+            if (rec->is_method) {
+                signatures += "self: " + self_type + ", ";
+            }
+            signatures += "*args, **kwargs) -> NotImplemented\n";
```

Comment:
Scrutinizing this again: `NotImplemented` is the value, and not the type, right. `type(NotImplemented)` prints `<class 'NotImplementedType'>` for me.

Also, is this the way it's done? Can you give me a reference of type annotations in pure Python code, perhaps?

Reply:
I think I had incorrectly assumed that being a singleton, that `NotImplemented` was treated like `None` and that you're allowed to supply it. In fact, mypy doesn't use or need it at all (https://github.com/python/mypy/issues/6710#issuecomment-485580032) and so it isn't often used. Given that comment, `Any` seems like the correct type to return.
Alternatively, we could combine the two signatures into one and simply say that the type of the second argument is always object, because it will always be accepted and mypy will assume that NotImplemented will be returned regardless. It's less useful for a user that way, but it resolves the original issue without much extra complexity in the implementation.